### PR TITLE
Workaround for broken TS getters (important for spreading State)

### LINKED ISSUE #511: Promise rejection issue in React Native
## Bug or feature request?

Bug in React Native, actually... sort of...

## Description:

I was getting an issue when a promise rejected, and the error I got was related to React Native's own polyfill of `Object.assign`. It fails on `hasOwnProperty`. I'm not sure if this is still an issue in the latest React Native, but I'm using expo so I'm a couple versions behind.

The offending code in RN is here: https://github.com/expo/react-native/blob/exp-latest/Libraries/polyfills/Object.es6.js#L54

The error I get is: "One of the sources for assign has an enumerable key on the prototype chain. Are you trying to assign a prototype property? We don't allow it, as this is an edge case that we do not support. This error is a performance optimization and not spec compliant."

For the record, the key it's breaking on is `inert` on a state object. Not sure what that's used for, but that's the one it doesn't like. This all takes place in the `resolveState` function.

I think this is related to the new spawning behavior in 4.6.x. I downgraded to 4.5.0, as I wasn't using the actor model stuff anyway and that fixed the issue for me. I don't really expect this library to address the weirdness in React Native, but I wanted to leave this comment here for posterity. I'm not even sure if it could be fixed from this end without breaking something else.

Maybe we can put it in a troubleshooting doc for any other poor souls that run into this?

### PR DESCRIPTION
**Pull request by Andarist**
fixes #511 

Not sure if we can do this in this major though. Was `.inert` getter part of the public API?

### CODE REVIEW

File: src/State.ts [hunk 1]
```diff
@@ -180,17 +174,8 @@ export class State<TContext, TEvent extends EventObject = EventObject>
...
-  /**
-   * The next events that will cause a transition from the current state.
-   */
-  public get nextEvents(): EventType[] {
-    if (!this.tree) {
-      return [];
-    }
 
-    return this.tree.nextEvents;
+    this.nextEvents = !config.tree ? [] : config.tree.nextEvents;
```
Comment (davidkpiano) [inline src/State.ts]: This is a potentially expensive operation.

Comment (Andarist) [reply src/State.ts]: Should I change it to a regular method then? Was thinking about removing it entirely, but that would force consumers to handle missing tree themselves, which is not that ergonomic.

Comment (davidkpiano) [reply src/State.ts]: Let's change it to a method, then. It's not public API.

**Comment by Andarist** [comment]
@davidkpiano changed `.nextEvents` to `.getNextEvents()` method

**Comment by davidkpiano** [comment]
Actually I misspoke, `state.nextEvents` is public API.

**Comment by Andarist** [comment]
Ok, I've brought back `.nextEvents` getter, but in such a form that it should fix the reported issue.

**Pull request merged**